addons/hr_expense/models/hr_expense.py

```diff
@@ -554,6 +554,8 @@ def _prepare_base_line_for_taxes_computation(self, **kwargs):
 
     def attach_document(self, **kwargs):
         """When an attachment is uploaded as a receipt, set it as the main attachment."""
+        if (not self.has_access('write') or (self.sheet_id and not self.sheet_id.has_access('write'))):
```

Comment:
```suggestion
        if not self.has_access('write') or (self.sheet_id and not self.sheet_id.has_access('write')):
```
If you're inlining it then those are not needed